# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/502_known_bugs.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/502_known_bugs.rs
--- a/third_party/rust/ron/tests/502_known_bugs.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/502_known_bugs.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,6 +1,9 @@
 use std::collections::{BTreeMap, HashMap};
 
-use ron::{error::Position, error::SpannedError, extensions::Extensions, ser::PrettyConfig, Error};
+use ron::{
+    error::Position, error::Span, error::SpannedError, extensions::Extensions, ser::PrettyConfig,
+    Error,
+};
 use serde::{Deserialize, Serialize};
 
 #[test]
@@ -39,7 +42,10 @@
                 field: "hi",
                 outer: None
             },
-            position: Position { line: 7, col: 2 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 2 }
+            }
         })),
     );
 }
@@ -95,7 +101,10 @@
                 field: "ho",
                 outer: Some(String::from("AdjacentlyTagged"))
             },
-            position: Position { line: 1, col: 58 }
+            span: Span {
+                start: Position { line: 1, col: 58 },
+                end: Position { line: 1, col: 58 }
+            }
         }),
     );
 }
@@ -135,7 +144,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 6, col: 2 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 2 }
+            }
         })),
     );
 }
@@ -182,7 +194,10 @@
                 field: "hi",
                 outer: None
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         })),
     );
 }
@@ -231,7 +246,10 @@
                 field: "hi",
                 outer: Some(String::from("C"))
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         })),
     );
 }
@@ -459,7 +477,10 @@
                 expected: String::from("an empty array"),
                 found: String::from("a unit value")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
     assert_eq!(
@@ -503,7 +524,10 @@
                 expected: String::from("i32"),
                 found: String::from("a sequence")
             },
-            position: Position { line: 5, col: 2 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 2 }
+            }
         }))
     );
 }
@@ -537,7 +561,10 @@
                 expected: String::from("i32"),
                 found: String::from("a sequence")
             },
-            position: Position { line: 1, col: 36 }
+            span: Span {
+                start: Position { line: 1, col: 36 },
+                end: Position { line: 1, col: 36 }
+            }
         })
     );
 }
@@ -559,7 +586,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 4, col: 2 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 2 }
+            }
         }))
     );
 }
@@ -594,7 +624,10 @@
                 expected: String::from("i32"),
                 found: String::from("a sequence")
             },
-            position: Position { line: 4, col: 1 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 1 }
+            }
         }))
     );
 }
@@ -631,7 +664,10 @@
                 expected: String::from("i32"),
                 found: String::from("a sequence")
             },
-            position: Position { line: 4, col: 1 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 1 }
+            }
         }))
     );
 }
@@ -672,7 +708,10 @@
                 expected: String::from("a tuple of size 1"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 6, col: 2 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 2 }
+            }
         }))
     );
 }
@@ -728,7 +767,10 @@
                 expected: String::from("an array of length 1"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 1, col: 79 }
+            span: Span {
+                start: Position { line: 1, col: 79 },
+                end: Position { line: 1, col: 79 }
+            }
         })
     );
 }
@@ -766,7 +808,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 5, col: 2 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 2 }
+            }
         }))
     );
 }
@@ -817,7 +862,10 @@
                 expected: String::from("an array of length 1"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -870,7 +918,10 @@
                 expected: String::from("a tuple of size 1"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -909,7 +960,10 @@
                 expected: String::from("tuple variant"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 7, col: 2 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 2 }
+            }
         }))
     );
 }
@@ -961,7 +1015,10 @@
                 expected: String::from("tuple variant"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 1, col: 50 }
+            span: Span {
+                start: Position { line: 1, col: 50 },
+                end: Position { line: 1, col: 50 }
+            }
         })
     );
 }
@@ -999,7 +1056,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 6, col: 2 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 2 }
+            }
         }))
     );
 }
@@ -1046,7 +1106,10 @@
                 expected: String::from("tuple variant"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -1095,7 +1158,10 @@
                 expected: String::from("tuple variant"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -1129,7 +1195,10 @@
                 expected: String::from("any valid RON-value-string"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 7, col: 2 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 2 }
+            }
         }))
     );
 }
@@ -1178,7 +1247,10 @@
                 expected: String::from("any valid RON-value-string"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 1, col: 39 }
+            span: Span {
+                start: Position { line: 1, col: 39 },
+                end: Position { line: 1, col: 39 }
+            }
         })
     );
 }
@@ -1211,7 +1283,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 6, col: 2 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 2 }
+            }
         }))
     );
 }
@@ -1255,7 +1330,10 @@
                 expected: String::from("any valid RON-value-string"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -1301,7 +1379,10 @@
                 expected: String::from("any valid RON-value-string"),
                 found: String::from("the unsigned integer `42`")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -1332,7 +1413,10 @@
                 expected: String::from("an empty array"),
                 found: String::from("a unit value")
             },
-            position: Position { line: 7, col: 2 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 2 }
+            }
         }))
     );
 }
@@ -1377,7 +1461,10 @@
                 expected: String::from("tuple struct TupleStruct"),
                 found: String::from("a unit value")
             },
-            position: Position { line: 1, col: 40 }
+            span: Span {
+                start: Position { line: 1, col: 40 },
+                end: Position { line: 1, col: 40 }
+            }
         })
     );
 }
@@ -1410,7 +1497,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 6, col: 2 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 2 }
+            }
         }))
     );
 }
@@ -1454,7 +1544,10 @@
                 expected: String::from("tuple variant"),
                 found: String::from("a unit value")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -1502,7 +1595,10 @@
                 expected: String::from("struct variant"),
                 found: String::from("a unit value")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -1545,7 +1641,10 @@
                 expected: String::from("any value"),
                 found: format!("integer `{}` as u128", i128::MAX)
             },
-            position: Position { line: 5, col: 52 }
+            span: Span {
+                start: Position { line: 5, col: 13 },
+                end: Position { line: 5, col: 52 }
+            }
         }))
     );
 }
@@ -1607,7 +1706,10 @@
                 expected: String::from("any value"),
                 found: format!("integer `{}` as u128", u128::MAX)
             },
-            position: Position { line: 1, col: 67 }
+            span: Span {
+                start: Position { line: 1, col: 28 },
+                end: Position { line: 1, col: 67 }
+            }
         }),
     );
 }
@@ -1650,7 +1752,10 @@
                 expected: String::from("any value"),
                 found: format!("integer `{}` as i128", i128::MIN)
             },
-            position: Position { line: 4, col: 53 }
+            span: Span {
+                start: Position { line: 4, col: 14 },
+                end: Position { line: 4, col: 53 }
+            }
         }))
     );
 }
@@ -1703,7 +1808,10 @@
                 expected: String::from("any value"),
                 found: format!("integer `{}` as u128", u128::MAX)
             },
-            position: Position { line: 4, col: 52 }
+            span: Span {
+                start: Position { line: 4, col: 13 },
+                end: Position { line: 4, col: 52 }
+            }
         }))
     );
 }
@@ -1758,7 +1866,10 @@
                 expected: String::from("any value"),
                 found: format!("integer `{}` as i128", i128::MIN)
             },
-            position: Position { line: 4, col: 53 }
+            span: Span {
+                start: Position { line: 4, col: 14 },
+                end: Position { line: 4, col: 53 }
+            }
         }))
     );
 }
@@ -1785,7 +1896,10 @@
                 field: "ho",
                 outer: None
             },
-            position: Position { line: 3, col: 9 }
+            span: Span {
+                start: Position { line: 3, col: 6 },
+                end: Position { line: 3, col: 9 }
+            }
         }))
     );
 }
@@ -1828,7 +1942,10 @@
                 field: "ho",
                 outer: Some(String::from("A"))
             },
-            position: Position { line: 3, col: 9 }
+            span: Span {
+                start: Position { line: 3, col: 6 },
+                end: Position { line: 3, col: 9 }
+            }
         }))
     );
 }
@@ -1852,7 +1969,10 @@
         ),
         Err(Err(SpannedError {
             code: Error::ExpectedString,
-            position: Position { line: 3, col: 5 }
+            span: Span {
+                start: Position { line: 3, col: 5 },
+                end: Position { line: 3, col: 5 }
+            }
         }))
     );
 }
@@ -1878,7 +1998,10 @@
         ),
         Err(Err(SpannedError {
             code: Error::ExpectedString,
-            position: Position { line: 3, col: 5 }
+            span: Span {
+                start: Position { line: 3, col: 5 },
+                end: Position { line: 3, col: 5 }
+            }
         }))
     );
 }
@@ -2008,7 +2131,10 @@
                 expected: String::from("i32"),
                 found: String::from("a map")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -2056,7 +2182,10 @@
                 expected: String::from("i32"),
                 found: String::from("a map")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -2163,7 +2292,10 @@
                 expected: String::from("i32"),
                 found: String::from("a unit value")
             },
-            position: Position { line: 4, col: 1 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 1 }
+            }
         }))
     );
 }
@@ -2198,7 +2330,10 @@
                 expected: String::from("i32"),
                 found: String::from("a map")
             },
-            position: Position { line: 6, col: 1 }
+            span: Span {
+                start: Position { line: 6, col: 1 },
+                end: Position { line: 6, col: 1 }
+            }
         }))
     );
 }
@@ -2235,7 +2370,10 @@
                 expected: String::from("i32"),
                 found: String::from("a sequence")
             },
-            position: Position { line: 7, col: 1 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 1 }
+            }
         }))
     );
 }
@@ -2271,7 +2409,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Unit\"")
             },
-            position: Position { line: 4, col: 1 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 1 }
+            }
         }))
     );
 }
@@ -2309,7 +2450,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Newtype\"")
             },
-            position: Position { line: 4, col: 1 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 1 }
+            }
         }))
     );
 }
@@ -2345,7 +2489,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Struct\"")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2381,7 +2528,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Unit\"")
             },
-            position: Position { line: 4, col: 1 }
+            span: Span {
+                start: Position { line: 4, col: 1 },
+                end: Position { line: 4, col: 1 }
+            }
         }))
     );
 }
@@ -2419,7 +2569,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Newtype\"")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2455,7 +2608,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Struct\"")
             },
-            position: Position { line: 7, col: 1 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 1 }
+            }
         }))
     );
 }
@@ -2493,7 +2649,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Tuple\"")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2529,7 +2688,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Flattened\"")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2567,7 +2729,10 @@
                 expected: String::from("i32"),
                 found: String::from("the string \"Flattened\"")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2586,7 +2751,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 1, col: 3 }
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 }
+            }
         }))
     );
 }
@@ -2605,7 +2773,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 1, col: 3 }
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 3 }
+            }
         }))
     );
 }
@@ -2655,7 +2826,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 3, col: 2 }
+            span: Span {
+                start: Position { line: 3, col: 1 },
+                end: Position { line: 3, col: 2 }
+            }
         }))
     );
 }
@@ -2687,7 +2861,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 3, col: 1 }
+            span: Span {
+                start: Position { line: 3, col: 1 },
+                end: Position { line: 3, col: 1 }
+            }
         }))
     );
 }
@@ -2718,7 +2895,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 3, col: 2 }
+            span: Span {
+                start: Position { line: 3, col: 1 },
+                end: Position { line: 3, col: 2 }
+            }
         }))
     );
 }
@@ -2752,7 +2932,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 3, col: 1 }
+            span: Span {
+                start: Position { line: 3, col: 1 },
+                end: Position { line: 3, col: 1 }
+            }
         }))
     );
 }
@@ -2793,7 +2976,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 3, col: 2 }
+            span: Span {
+                start: Position { line: 3, col: 1 },
+                end: Position { line: 3, col: 2 }
+            }
         }))
     );
 }
@@ -2834,7 +3020,10 @@
                 expected: String::from("unit"),
                 found: String::from("a map")
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2880,7 +3069,10 @@
             code: Error::Message(String::from(
                 "data did not match any variant of untagged enum Untagged"
             )),
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2920,7 +3112,10 @@
                 expected: String::from("map with a single key"),
                 found: String::from("a map"),
             },
-            position: Position { line: 5, col: 1 }
+            span: Span {
+                start: Position { line: 5, col: 1 },
+                end: Position { line: 5, col: 1 }
+            }
         }))
     );
 }
@@ -2960,7 +3155,10 @@
                 expected: String::from("map with a single key"),
                 found: String::from("a map"),
             },
-            position: Position { line: 7, col: 1 }
+            span: Span {
+                start: Position { line: 7, col: 1 },
+                end: Position { line: 7, col: 1 }
+            }
         }))
     );
 }
@@ -3060,7 +3258,7 @@
     );
 }
 
-fn check_roundtrip<T: PartialEq + std::fmt::Debug + Serialize + serde::de::DeserializeOwned>(
+fn check_roundtrip<T: PartialEq + core::fmt::Debug + Serialize + serde::de::DeserializeOwned>(
     val: &T,
     config: PrettyConfig,
 ) -> Result<(), Result<Error, SpannedError>> {